add atomicPhysicsCI setup (#5129)
ComputationalRadiationPhysics/picongpu: share/picongpu/examples/AtomicPhysics/README.md

Atomic Physics
==============

.. sectionauthor:: Brian Marre, Tapish Narwal

This example demonstrates the use of the atomic physics[AtomicPhysics vs Ionization]_ and is intended as a starting point for simulations using atomicPhysics(FLYonPIC).

This setup models a homogeneous charge neutral copper plasma with an ion density of 2e28 1/m^3 and an initial copper charge state of 2.

> Copper ions are initialized cold, no initial momentum, in the ground state, while electrons are given an initial Maxwellian-Temperature of 1 keV.

Reviewer: ```suggestion
Copper ions are initialized cold, with no initial momentum, in the ground state, while electrons are given an initial Maxwellian-Temperature of 1 keV.
```

Author: done

.. warning::

     This setup does not include elastic collisions

The copper atomic state distribution's initial state will relax over time towards equilibrium.

.. warning::

     The exact equilibrium is dependent on the atomic physics model.

     Therefore direct comparisons of different atomic models should not rely on the equilibrium model only.
     This is especially true if comparing to SCFLY/FlyCHK since they include recombination paths which are not yet implemented in atomicPhysics(FLYonPIC)

Atomic Input Data
-----------------

Any simulation using atomicPhysics(FLYonPIC) requires a user provided set of atomic input data files, these files are not provided with PIConGPU due to licensing issues.

> This simulation is intended to use the atomic input data files from `https://github.com/ComputationalRadiationPhysics/FLYonPICAtomicTestData`.

Reviewer: Should we link to a private repository?

Author: Well so long as it is private the URL will not resolve unless the user has logged in and the user is authorized.
And it's the easiest way to distribute the input files to users.

> If you have access to this repository by `source ./etc/getAtomicInputData.sh` to copy a set of atomic input files to the `atomicInputData`-directory before running the simulation.

Reviewer: ```suggestion
If you have access to this repository, execute `./etc/getAtomicInputData.sh` to copy a set of atomic input files to the `atomicInputData`-directory before running the simulation.
```

Author: done

.. note::

     compilation alone does not require atomic input data files

Running CI runtime test for atomicPhysics
-----------------------------------------

This setup is also a run-time integration tests of FLYonPIC.

To build the run-time test use the following steps:
1.) copy the atomic physics example using `pic-create` to a new setup folder
2.) go inside the new setup folder

> 2.) provide the atomic input data using `source ./etc/getAtomicInputData.sh`, requires access to the test setup.

Reviewer: ```suggestion
2.) provide the atomic input data using `./etc/getAtomicInputData.sh`, requires access to the test setup.
```
Shouldn't source unless required

Author: done

3.) build the setup using either `./etc/build.sh` or by hand using `pic-build -c "-DPARAM_FORCE_CONSTANT_ELECTRON_TEMPERATURE=true"`.

To run the test supplied with this use one of the three ways described below.
 - run locally
 - run using tbg
 - run using the JUBE CI workflow

Run Locally
___________

From the root directory
1.) run picongpu by hand using `./bin/picongpu -g 1 1 1 -d 1 1 1 -s 25 --periodic 1 1 1`

> 2.) run `source ./validate/validateLocally.sh` to validate the simulation results

Reviewer: ```suggestion
2.) run `./validate/validateLocally.sh` to validate the simulation results
```
Shouldn't source unless required

Author: done

Run Using TBG
_____________

> The setup comes with `.cfg`-files designed to run on NVIDIA A100 (40 GB) GPU nodes from the JUBE CI.

Reviewer: ```suggestion
The setup comes with `.cfg`-files designed to run on NVIDIA A100 (40 GB) GPU nodes from JURECA using the JUBE CI.
```

Author: done
